docs/spec/sources.md

```diff
@@ -362,5 +375,41 @@ _Structured Content Mode_ of the HTTP Protocol Binding for CloudEvents. Sources
 SHOULD send events to its
 [Destination](https://pkg.go.dev/github.com/knative/pkg/apis/duck/v1?tab=doc#Destination).
 
+### CloudEvent `knsource` extension
+
+Sources SHOULD produce CloudEvents with the `knsource` CloudEvent extension
+identifying the originating [Source Custom Object](#source-custom-objects). It
+is highly-recommended producing the `knsource` CloudEvent extension.
```

Comment:
```suggestion
is highly-recommended to produce the `knsource` CloudEvent extension.
```

Maybe we should add a note why, e.g. _"So that subscribers can consume events originating from a specific source instance."_

It's kind of explained in the last paragraph of this section but both paragraphs feel a bit disconnected to me.

Reply:
I'll blame the stupid grammar checker. 

The reason I put the use case at the end is to make the spec as clear and concise as possible. Maybe explicitly stating it's a use case may help?

Reply:
I think it would help to have it in the same sentence so the reader immediately understands _why_, otherwise the explanation comes much later.

cc @devguyio who suggested the same change pretty much.